src/Silicon.cpp

```diff
@@ -275,31 +375,52 @@ namespace galsim {
         const int step = target.getStep();
         const int stride = target.getStride();
 
-        std::vector<bool> changed(nx * ny, false);
+        int nxny = nx * ny;
+        
+        int imageDataSize = target.getMaxPtr() - target.getData();
+        T* targetData = target.getData();
 
+        Position<float>* horizontalBoundaryPointsData = _horizontalBoundaryPoints.data();
+        Position<float>* verticalBoundaryPointsData = _verticalBoundaryPoints.data();
+        Position<float>* horizontalDistortionsData = _horizontalDistortions.data();
+        Position<float>* verticalDistortionsData = _verticalDistortions.data();
+
+        bool* changedData = _changed.get();
+        
         // Loop through the boundary arrays and update any points affected by nearby pixels
         // Horizontal array first
-        const T* ptr = target.getData();
+        // map image data and changed array throughout all GPU loops
+
 #ifdef _OPENMP
+#ifndef GALSIM_USE_GPU
 #pragma omp parallel for
+#else
+#pragma omp target teams distribute parallel for
+#endif
 #endif
-        for (int p=0; p < (ny * nx); p++) {
+        for (int p=0; p < nxny; p++) {
             // Calculate which pixel we are currently below
             int x = p % nx;
             int y = p / nx;
 
             // Loop over rectangle of pixels that could affect this row of points
-            int polyi1 = std::max(x - _qDist, 0);
-            int polyi2 = std::min(x + _qDist, nx - 1);
+            // std::min and std::max are causing this loop to crash, even though
+            // the same functions run fine in the accumulate loop.
+            int polyi1 = x - _qDist;
```

Comment:
I think the versions with min() and max() are clearer.  Would a macro a la https://stackoverflow.com/a/3437484 work?  (And same comment throughout this file...)